Framework/DataHandling/test/SetBeamTest.h

```diff
@@ -0,0 +1,120 @@
+    inputWS->setInstrument(testInst);
+
+    auto alg = createAlgorithm();
+    TS_ASSERT_THROWS_NOTHING(alg->setProperty("InputWorkspace", inputWS));
+    alg->setProperty("Geometry", createRectangularBeamProps());
+    TS_ASSERT_THROWS_NOTHING(alg->execute());
+
+    auto source = testInst->getSource();
+    const auto &instParams = inputWS->constInstrumentParameters();
+    auto beamWidth = instParams.get(source->getComponentID(), "beam-width");
+    TS_ASSERT(beamWidth);
+    const double widthValue = beamWidth->value<double>();
+    TS_ASSERT_DELTA(0.01, widthValue, 1e-10);
+    auto beamHeight = instParams.get(source->getComponentID(), "beam-height");
+    TS_ASSERT(beamHeight);
+    const double heightValue = beamHeight->value<double>();
+    TS_ASSERT_DELTA(0.0075, heightValue, 1e-10);
+  }
+
+  //----------------------------------------------------------------------------
+  // Failure tests
+  //----------------------------------------------------------------------------
+  void test_Workspace_Without_Instrument_Not_Accepted() {
+    auto inputWS = WorkspaceCreationHelper::Create2DWorkspaceBinned(1, 1);
+
+    auto alg = createAlgorithm();
+    TS_ASSERT_THROWS(alg->setProperty("InputWorkspace", inputWS),
+                     std::invalid_argument);
+  }
+
+  void test_No_Geometry_Inputs_Not_Accepted() {
+    auto inputWS = WorkspaceCreationHelper::Create2DWorkspaceBinned(1, 1);
+    auto testInst = ComponentCreationHelper::createTestInstrumentCylindrical(1);
+    inputWS->setInstrument(testInst);
+
+    auto alg = createAlgorithm();
+    alg->setProperty("InputWorkspace", inputWS);
+    TS_ASSERT_THROWS(alg->execute(), std::runtime_error);
+  }
+
+  void test_Missing_Geometry_Inputs_Not_Accepted() {
+    using Mantid::Kernel::PropertyManager;
+    auto inputWS = WorkspaceCreationHelper::Create2DWorkspaceBinned(1, 1);
+    auto testInst = ComponentCreationHelper::createTestInstrumentCylindrical(1);
+    inputWS->setInstrument(testInst);
+
+    auto alg = createAlgorithm();
+    alg->setProperty("InputWorkspace", inputWS);
+    auto props = boost::make_shared<PropertyManager>();
+    alg->setProperty("Geometry", props);
+    TS_ASSERT_THROWS(alg->execute(), std::runtime_error);
+    props = createRectangularBeamProps();
+    props->removeProperty("Width");
+    alg->setProperty("Geometry", props);
+    TS_ASSERT_THROWS(alg->execute(), std::runtime_error);
+    props->removeProperty("Height");
+    alg->setProperty("Geometry", props);
+    TS_ASSERT_THROWS(alg->execute(), std::runtime_error);
+  }
+
+  //----------------------------------------------------------------------------
+  // Non-test methods
+  //----------------------------------------------------------------------------
+private:
+  Mantid::API::IAlgorithm_uptr createAlgorithm() {
+    auto alg = Mantid::Kernel::make_unique<SetBeam>();
+    alg->setChild(true);
+    alg->setRethrows(true);
+    alg->initialize();
+    return std::move(alg);
```

Comment:
Won't the default be to copy as alg is an lvalue? I'm pretty sure the compiler complained when I didn't have it.